### PR TITLE
adding depth_test control to Markers visual. Fixes #1060 .

### LINKED ISSUE #1060: Sphere (CompoundVisual) does not get opaque to markers
Using this example code, and rotating the camera around the scene until the markers are behind the sphere, the markers always stay visible, although I tried to make an opaque sphere. For data exploration, this is highly confusing, so I'm looking for a way to make the sphere visual opaque?
Note, I also tried a RGBA sphere color like (0,0,1,1), but to no avail.

``` python
import sys
import numpy as np
from vispy import scene
from vispy.scene import visuals

# create particle data

z = np.linspace(0.8, 2, 100)
x = np.zeros_like(z)
y = np.zeros_like(z)

data = np.vstack([x, y, z]).T
#
# Make a canvas and add simple view
#
canvas = scene.SceneCanvas(keys='interactive', show=True, bgcolor='w')

view = canvas.central_widget.add_view()
# view.camera = 'turntable'
view.camera = 'arcball'
view.camera.fov = 60

# create Enceladus
enc = visuals.Sphere(radius=0.8, color=(0, 0, 1),
                     edge_color=(1, 0, 0), parent=view.scene)

# add a colored 3D axis for orientation
axis = visuals.XYZAxis(parent=view.scene, width=1)

scatter = visuals.Markers()

scatter.set_data(data, edge_color=None, face_color=(0.6, 0.5, 0.4), size=5)

view.add(scatter)


if __name__ == '__main__' and sys.flags.interactive == 0:
    canvas.app.run()
```

### PR DESCRIPTION
**Pull request by michaelaye**
The markers (jet particles) are now hiding behind Enceladus, as it should be. ;)

screenshot 2015-08-17 13 50 39

### CODE REVIEW

**Comment by michaelaye** [comment]
That's a weird fail in Travis, anybody knows what's going on there?

**Comment by rougier** [comment]
Some Travis hiccup. I don't see anything else. Can you do a dummy commit just to start travis again ?

**Comment by Eric89GXL** [comment]
I'll just manually restart

File: vispy/visuals/markers.py [hunk 1]
```diff
@@ -480,8 +480,14 @@
 
 class MarkersVisual(Visual):
     """ Visual displaying marker symbols.
+
+    Parameters
+    ----------
+    depth_test : boolean
+        Control if the markers are checked for being visible.
```
Comment (Eric89GXL) [inline vispy/visuals/markers.py]: can you add something for `**kwargs` as well?

**Comment by michaelaye** [comment]
how's that?

**Comment by michaelaye** [comment]
If you want, I actually can go through all the visuals to see about their docstring status, I found a hickup or two. Just make one PR for that would be okay?

**Comment by Eric89GXL** [comment]
Good except for the errant extra newline between parameters.

You can include the other hiccup fixes here if you want, this PR is small enough :)

**Comment by michaelaye** [comment]
Ok, do you in general want defaults to be documented?

**Comment by Eric89GXL** [comment]
I don't think we've settled on a standard for that, but I think most other packages are doing it nowadays. So sure.

**Comment by michaelaye** [comment]
I shied away from doing the default docstrings as it's already messy with just fixing the **kwargs. It was mentioned in some docstrings but not actually implemented and vice versa. Think I added it always to the obvious superclass, but maybe sometimes the set_data method could also use it? Not 100% sure about that for all cases.

**Comment by Eric89GXL** [comment]
@michaelaye you need a rebase

File: vispy/visuals/image.py [hunk 2]
```diff
@@ -239,13 +244,13 @@ def _build_vertex_data(self):
         mgrid[..., 1] *= h
 
         quads[..., :2] += mgrid
-        tex_coords = quads.reshape(grid[1]*grid[0]*6, 3)
+        tex_coords = quads.reshape(grid[1] * grid[0] * 6, 3)
```
Comment (Eric89GXL) [inline vispy/visuals/image.py]: FYI PEP8 E226 (whitespace around operator) is considered optional. I don't think `numpy` and `scipy` even enforce it, and we don't. You can keep the change here, though.

**Comment by Eric89GXL** [comment]
@michaelaye if you want a fun challenge, you can try to add `**kwarg` detection in our automated docstring checker. It would have found a bunch of those instances you manually searched for / came across:

https://github.com/vispy/vispy/blob/master/vispy/util/tests/test_docstring_parameters.py

File: vispy/visuals/image.py (hunk 2 above)
Comment (michaelaye) [inline vispy/visuals/image.py]: yes, thanks. Did not have it deactivated in my editor.

**Comment by liubenyuan** [comment]
I also vote for adding `depth_test=True` in `MarkersVisual` for 3D scatter plot. #1191

**Comment by Eric89GXL** [comment]
@michaelaye is this good to go from your end, other than the rebase?

**Comment by astrofrog** [comment]
@michaelaye - thanks for working on this! This solves a long-standing problem in the 3D viewer for glue (https://github.com/glue-viz/glue-vispy-viewers/issues/114). I just tested it out, and it works great.

**Comment by astrofrog** [comment]
One thing I have noticed though is that if I set ``depth_test=True`` for markers, the edge color is ignored (and set to white in my case).

### No depth test:

screen shot 2016-05-18 at 5 54 36 pm

### With depth test:

screen shot 2016-05-18 at 5 53 18 pm

Any idea what this could be due to?

**Comment by Eric89GXL** [comment]
@astrofrog let's not release until this is in, then. @michaelaye let us know if you have time to continue working on this, otherwise one of us be able to take it over to get it mergable in the next couple of weeks.

@rougier have you seen this `depth_test=True` problem with markers @astrofrog sees?

**Comment by Eric89GXL** [comment]
@astrofrog looks like that depth test issue has already been raised in #1160

**Comment by Eric89GXL** [comment]
Looks like this was fixed in another PR recently, feel free to open a new PR if it isn't fixed right for you @michaelaye

**Comment by michaelaye** [comment]
Thanks! Was both the depth_test and the edge color issue fixed there?

**Comment by michaelaye** [comment]
hm, i don't find the PR fixing this, where is it? It's not in #1160

**Comment by Eric89GXL** [comment]
Oh sorry, I meant that the edge issue is already in #1160, and remains unfixed.

`depth_test=True` was added here:

https://github.com/vispy/vispy/commit/393334e1f79996ec5e14a5c1eeced47fbabcc27d

It was fixed to `True` unlike in your PR where it was an option, so feel free to open another PR (or I'll reopen this one if you plan to rebase) if you'd rather have it be an issue.

**Pull request closed without merging**